sktime/utils/_testing/annotation.py

```diff
@@ -10,7 +16,17 @@ def make_annotation_problem(
     make_X=False,
     n_columns=2,
     random_state=None,
+    estimator_type=None,
 ):
+    if estimator_type == "Poisson":
+        rng = check_random_state(random_state)
+        y = piecewise_poisson(
+            lambdas=rng.randint(1, 4, n_timepoints),
+            lengths=rng.randint(1, 5, n_timepoints),
+            random_state=random_state,
+        )
+        return y
```

Comment:
Its worth thinking about what might happen in a case where we want a poisson distributed `y` but also `X`.  What if we wanted a poisson distributed `X`?  I think it is ok that we do not cover all of these cases here, and I don't think that writing things this way should break anything yet (since we are typically not thinking about `X` at all for the `PoissonHMM` anyway.  I was just thinking it could be useful to leave a comment to this affect! :)

Reply:
We gave the possibility of a poisson distributed `X` some thought and decided to not implement it in this case as we thought it was sufficient to only have a poisson distributed `y`. However, we do have the following code below for the request if needed. 

```
if estimator_type == "Poisson":
        rng = check_random_state(random_state)
        y = piecewise_poisson(
            lambdas=rng.randint(1, 4, n_timepoints),
            lengths=rng.randint(1, 5, n_timepoints),
            random_state=random_state,
        )
        if not make_X:
            return y
            
        X = piecewise_poisson(
            lambdas=rng.randint(1, 4, n_timepoints),
            lengths=rng.randint(1, 5, n_timepoints),
            random_state=random_state,
        )
        return y, X
```

Reply:
I tend to agree with your hunch - ie that it probably isn't necessary and is just contributing the code bloat.  It also isn't clear to me that one would necessarily want to for X to also be poisson in this case that we want y to be poisson, so I wouldn't necessarily say we should default to the code you suggested.  Perhaps a better compromise would be to raise a ValueError if someone passes in both `make_X` and `estimator_type` = "Poisson".  That would give someone an informative error if they do run into this edge case int he future, and also serve as a potential warning to future devs that this edge case isn't covered.

Reply:
Thanks @miraep8, 

We have added the following logic to the condition

```
    if estimator_type == "Poisson":
        if make_X:
            raise ValueError("PoissonHMM creates a distribution for y only")
```

Reply:
Nice!